homeassistant/components/peco/sensor.py

```diff
@@ -0,0 +1,117 @@
+from peco import BadJSONError, HttpError, InvalidCountyError
+
+from homeassistant.components.sensor import SensorEntity, SensorStateClass
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import _LOGGER, DOMAIN, SCAN_INTERVAL
+
+PARALLEL_UPDATES: Final = 0
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the sensor platform."""
+    api = hass.data[DOMAIN][config_entry.entry_id]["api"]
+    websession = hass.data[DOMAIN][config_entry.entry_id]["websession"]
+    conf: MappingProxyType[str, Any] = config_entry.data
+
+    async def async_update_data():
+        """Fetch data from API."""
+        try:
+            async with async_timeout.timeout(10):
+                if conf["county"] == "TOTAL":
+                    return await api.get_outage_totals(websession)
+                else:
+                    return await api.get_outage_count(conf["county"], websession)
+        except InvalidCountyError as err:
+            raise ConfigEntryAuthFailed from err
+        except HttpError as err:
+            raise UpdateFailed from err
+        except BadJSONError as err:
+            raise UpdateFailed from err
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="PECO Outage Count",
+        update_method=async_update_data,
+        update_interval=timedelta(minutes=SCAN_INTERVAL),
+    )
+
+    await coordinator.async_config_entry_first_refresh()
+
+    _LOGGER.info("Setting up sensor platform")
+    county = conf["county"]
+    _LOGGER.info("County: %s", county)
+    async_add_entities(
+        [PecoOutageCounterSensorEntity(hass, config_entry.title, county, coordinator)],
+        True,
+    )
+    return
+
+
+class PecoOutageCounterSensorEntity(CoordinatorEntity, SensorEntity):
+    """PECO outage counter sensor."""
+
+    _attr_state_class = SensorStateClass.MEASUREMENT
+    _attr_icon: str = "mdi:power-plug-off"
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        name: str,
+        county: str,
+        coordinator: DataUpdateCoordinator,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self.hass = hass
+        self._county = county
+        self._name = name
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def native_value(self) -> int:
+        """Return the value of the sensor."""
+        return self.coordinator.data["customers_out"]
+
+    @property
+    def extra_state_attributes(self):
```

Comment:
Using `extra_state_attributes` should be avoided. Consider creating entities for these values instead.
Best practice would be to make use of SensorEntityDescriptions
https://github.com/home-assistant/core/blob/b05b4c4b3836cce84a63de0f5746ed231a1166cf/homeassistant/components/sensor/__init__.py#L220